src/cacheExchange.ts

```diff
@@ -103,6 +120,13 @@ export const cacheExchange = (opts?: CacheExchangeOpts): Exchange => ({
     opts.keys
   );
 
+  let hydration: void | Promise<void>;
+  if (opts.storage) {
+    hydration = opts.storage.read().then(data => {
```

Comment:
I opted to keep this on the store since it kinda belongs there more than in the data folder